src/MPDeviceBleImpl.cpp

```diff
@@ -766,6 +766,32 @@ void MPDeviceBleImpl::nihmReconditioning()
     mpDev->enqueueAndRunJob(jobs);
 }
 
+void MPDeviceBleImpl::getSecurityChallenge(const QString &key, const MessageHandlerCb &cb)
+{
+    AsyncJobs *jobs = new AsyncJobs("Send security challenge to device", this);
+
+    jobs->append(new MPCommandJob(mpDev, MPCmd::AUTH_CHALLENGE,
+                                Common::toHexArray(key),
+                                [this, cb](const QByteArray &data, bool &)
+    {
+        if (bleProt->getFirstPayloadByte(data) == MSG_FAILED )
```

Comment:
what if the first payload byte is also MSG_FAILED but the message length is longer?

Reply:
it is a good point, changing it to check the size of the response